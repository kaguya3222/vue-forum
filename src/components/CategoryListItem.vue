<template>
  <div class="forum-list">
    <h2 class="list-title">
      <router-link :to="{ name: 'Category', params: { id: category['.key'] } }">
        {{ category.name }}
      </router-link>
    </h2>
    <forum-list :forums="categoryForums" />
  </div>
</template>

<script>
import ForumList from "./ForumList";
import { mapGetters } from "vuex";

export default {
  name: "CategoryListItem.vue",
  components: { "forum-list": ForumList },
  props: {
    category: {
      required: true,
      type: Object
    }
  },
  computed: {
    ...mapGetters({
      forums: "forums/items"
    }),
    categoryForums() {
      return Object.values(this.forums).filter(
        forum => forum.categoryId === this.category[".key"]
      );
    }
  }
};
</script>

<style scoped></style>
