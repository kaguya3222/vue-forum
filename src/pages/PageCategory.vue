<template>
  <div class="col-full col-full_centered">
    <div class="col-full push-top">
      <h1>{{ category.name }}</h1>
    </div>
    <category-list-item :category="category" />
  </div>
</template>

<script>
import CategoryListItem from "../components/CategoryListItem";
export default {
  props: {
    id: {
      required: true,
      type: String
    }
  },
  name: "PageCategory",
  components: { "category-list-item": CategoryListItem },
  computed: {
    category() {
      return this.$store.state.categories[this.id];
    }
  }
};
</script>

<style scoped></style>
