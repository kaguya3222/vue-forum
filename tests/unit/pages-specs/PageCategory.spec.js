import { shallowMount } from "@vue/test-utils";
import PageCategory from "../../../src/pages/PageCategory";
import CategoryListItem from "../../../src/components/CategoryListItem";
import mockedSourceData from "../mocks/mockedSourceData";

describe("PageCategory", () => {
  const wrapper = shallowMount(PageCategory, {
    propsData: {
      id: mockedSourceData.categories[0][".key"]
    }
  });
  test("Shows category data on category page", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("Accepts CategoryListItem as a child component", () => {
    expect(wrapper.contains(CategoryListItem)).toBe(true);
  });
  test("Takes category object from source data with given id", () => {
    expect(wrapper.vm.category[".key"]).toBe(wrapper.vm.id);
  });
});
