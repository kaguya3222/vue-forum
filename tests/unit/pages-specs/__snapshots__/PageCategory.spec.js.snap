// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageCategory Shows category data on category page 1`] = `
<div class="col-full col-full_centered">
  <div class="col-full push-top">
    <h1>Feedback &amp; Information</h1>
  </div>
  <category-list-item-stub category="[object Object]"></category-list-item-stub>
</div>
`;
